priv/perf/apps/load_test/lib/common/generators.ex

```diff
@@ -0,0 +1,128 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule LoadTest.Common.Generators do
+  @moduledoc """
+  Provides helper functions to generate bundles of various useful entities for performance tests
+  """
+
+  alias ExPlasma.Encoding
+  alias LoadTest.ChildChain.Transaction
+  alias LoadTest.Ethereum
+  alias LoadTest.Ethereum.Account
+
+  @generate_user_timeout 600_000
+
+  @doc """
+  Creates addresses with private keys and funds them with given `initial_funds_wei` on geth.
+
+  Options:
+    - :faucet - the address to send the test ETH from, assumed to be unlocked and have the necessary funds
+    - :initial_funds_wei - the amount of test ETH that will be granted to every generated user
+  """
+  @spec generate_users(non_neg_integer) :: [OMG.TestHelper.entity()]
+  def generate_users(size) do
+    1..size
+    |> Task.async_stream(fn _ -> generate_user() end, timeout: @generate_user_timeout)
```

Comment:
the problem with this test is that they only run towards geth or ganache in ci or locally, and because of that, it makes little sense to call them performance tests.  they cannot give any reliable information of the performance of our system because they're not being run in isolation, on the same hardware, with vacuum setup.
so the question is, what value do these tests give us? I think it's very limited if not harmful because they take engineering hours and CPU cycles (running them, maintaining them).

@achiurizo

Reply:
I think most of the tests in perf (if not all) are running against geth

Reply:
yeah, that's probably true. if we want to make them useful in any way, it makes sense to review them against the existing perf tests and adapt them into the same "form", so that we can use infura or geth to stress test!